scripts/iso_mirna_quantification.py

```diff
@@ -0,0 +1,337 @@
+        the new directories and/or None
+    """
+    mir_out = None
+    isomir_out = None
+    iso_mirna_out = None
+
+    if 'mirna' in mir_list:
+        mir_out = outdir/f'mirna_counts_{library}'
+    if 'isomir' in mir_list:
+        isomir_out = outdir/f'isomir_counts_{library}'
+    if 'iso_mirna' in mir_list:
+        iso_mirna_out = outdir/f'iso_mirna_counts_{library}'
+
+    return mir_out, isomir_out, iso_mirna_out
+
+
+def get_contribution(aln: pysam.AlignedSegment, collapsed: bool = False, nh: bool = False) -> float:
+    """Get contribution of an alignment to the overall count.
+
+    Calculate the contribution of the alignment as the number of reads divided
+    by NH. The arguments `collapsed` and `nh` determine if these values are
+    taken from the alignment name. If both arguments are set to `True`, the
+    contribution of each alignment is computed as the number of reads/NH and
+    the values are extracted from the name. If only `collapsed` is `True`, the
+    contribution is the number of reads/NH tag. If only `nh` is `True`, the
+    contribution is 1/NH. Otherwise, the contribution is 1/NH tag. If `nh` is
+    `False` and the alignment misses the NH tag, the value is set to 1.
+
+    Args:
+        aln:
+            alignment to calculate the contribution from
+        collapsed:
+            indicates if the alignment is collapsed or not
+        nh:
+            indicates if the NH vale is found on the alignment name
+    Returns:
+        the contribution of the alignment to the overall total
+    """
+    query_id = aln.query_name
+
+    if collapsed and nh:
+        num_reads = int(query_id.split('-')[1].split('_')[0])
+        nh_value = int(query_id.split('-')[1].split('_')[1])
+
+    elif not collapsed and nh:
+        num_reads = 1
+        nh_value = int(query_id.split('_')[1])
+
+    elif collapsed and not nh:
+        num_reads = int(query_id.split('-')[1])
+        try:
+            nh_value = aln.get_tag("NH")
+        except KeyError:
+            nh_value = 1
+
+    else:
+        num_reads = 1
+        try:
+            nh_value = aln.get_tag("NH")
+        except KeyError:
+            nh_value = 1
+
+    return num_reads/nh_value
+
+
+def get_name(pre_name: str) -> list[str]:
+    """Get the final name for the spieces name.
+
+    Take a string and processes it to obtain the final name for the species
+    and the type of miRNA the string belongs to. Only the feat_name is
+    returned if the 3p-shift and 5p-shift are 0 and the CIGAR and MD are the
+    same - excluding the last character in the CIGAR string. Otherwise, the
+    whole input string is returned.
+
+    Args:
+        pre_name:
+            string with the format feat_name|5p-shift|3p-shift|CIGAR|MD
+
+    Returns:
+        list with the species name to be found in the final table and its type
+    """
+    data_name = pre_name.split("|")
+
+    if data_name[1] == '0' and data_name[2] == '0' and data_name[3][:2] == data_name[4]:
```

Comment:
I think the CIGAR and MD check is really lazy. You should check them fully and individually, then compare the number for equality. And don't rely on their length to be always double digit. Even if it's going to be always true (which you never know), I think it's inelegant to hardcode such assumptions.